task_manager/task_manager.cc

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "task_manager.hh"
+#include "db/config.hh"
+
+seastar::gate& task_manager::async_gate() {
+    return _gate;
+}
+
+abort_source& task_manager::abort_source() {
+    return _as;
+}
+
+std::chrono::seconds task_manager::task_ttl() const {
+    return std::chrono::seconds(_cfg.task_ttl_in_seconds());
+}
+
+task_manager::tasks_per_module& task_manager::get_tasks_per_module() {
+    return _tasks_per_module;
+}
+
+task_manager::task_map& task_manager::get_all_tasks() {
+    return _all_tasks;
+}
+
+task_manager::task_ptr task_manager::get_task_this_shard(task_id id) const {
```

Comment:
per-shard scope is the default for sharded service, so this need not be represented in the function name.
See replica/database.hh for example for global functions, that are defined as static, get the `sharded<replica::database>` as parameter, and are called `*_on_all_shards`.

Reply:
Also, I'm not sure if this method is really needed, it should be ok to open code this at the use site, that can call `get_all_tasks()` and search it.

That said, we should probably add a const-qualified variant of `get_all_tasks` and `get_tasks_per_module` (defined in the header so they'll get inlined)